workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/AdditiveViewFactory.kt

```diff
@@ -0,0 +1,84 @@
+package com.squareup.workflow1.ui
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import kotlin.reflect.KClass
+
+/**
+ * A [ViewFactory] for [OuterT] that delegates to the one for [InnerT].
+ * Makes it convenient for [OuterT] to wrap instances of [InnerT] to
+ * add information or behavior, without requiring wasteful wrapping
+ * in the view system.
+ *
+ * @param map called to convert instances of [OuterT] to [InnerT], and to
+ * allow [ViewEnvironment] to be transformed.
+ *
+ * @param initView called after the [ViewFactory] for [InnerT] has created a [View].
+ * Defaults to a no-op. Note that the [ViewEnvironment] is accessible via [View.environment].
+ *
+ * @param doShowRendering called to apply the [ViewShowRendering] function for
+ * [InnerT], allowing pre- and post-processing. Default implementation simply
+ * applies [map] and makes the function call.
+ */
+@WorkflowUiExperimentalApi
+class AdditiveViewFactory<OuterT : Any, InnerT : Any>(
```

Comment:
Just had a thought - what about `DecorativeViewFactory` as it harkens to the `Decorator` pattern?

Reply:
I like that. Or `DecoratorViewFactory`.

Reply:
+1 for decorative!